src/Language/PureScript/AST/SourcePos.hs

```diff
@@ -38,13 +38,16 @@ displaySourcePosShort sp =
     ":" <> T.pack (show (sourcePosColumn sp))
 
 instance A.ToJSON SourcePos where
-  toJSON SourcePos{..} =
-    A.toJSON [sourcePosLine, sourcePosColumn]
+  toJSON = \case
+    NullSourcePos -> A.Null
```

Comment:
This is a breaking change in that anything consuming this JSON now needs to handle the `null` case where it wouldn't have had to previously. If this data type occurs in CoreFn, that will be a problem. I think this is worse than losing the roundtrip property. We should have the roundtrip property for JSON produced by the current version, but losing it for JSON that can only be produced by previous versions is fine, I think.